include/aws/common/platform.h

```diff
@@ -0,0 +1,29 @@
+#ifndef AWS_COMMON_PLATFORM_H
+#define AWS_COMMON_PLATFORM_H
+
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/common/config.h>
+
+#ifdef _WIN32
+#    define AWS_OS_WINDOWS
+#elif __APPLE__
+#    define AWS_OS_APPLE
```

Comment:
There's other code in aws-c-common ifdefing on `_WIN32 __APPLE__ and __linux__`
Maybe go replace it with these macros?

Also see some in aws-c-io and aws-c-auth...

Reply:
I'll do that in a follow-up, this is to unblock other teams, so I need this first.